pkg/controller/nodemaintenance/nodemaintenance_controller.go

```diff
@@ -150,12 +182,17 @@ func (r *ReconcileNodeMaintenance) Reconcile(request reconcile.Request) (reconci
 			}
 		}
 	} else {
+		reqLogger.Infof("Deletion timestamp not zero")
+
 		// The object is being deleted
 		if ContainsString(instance.ObjectMeta.Finalizers, nodemaintenanceapi.NodeMaintenanceFinalizer) {
-			// Stop node maintenance - uncordon and remove live migration taint from the node.
-			if err := r.stopNodeMaintenance(instance.Spec.NodeName); err != nil {
-				if errors.IsNotFound(err) == false {
-					return r.reconcileAndError(instance, err)
+			if instance.Status.ErrorOnLeaseCount == 0 {
```

Comment:
I think the old code was correct.  It should be possible to run `stopNodeMaintenance()` multiple times without error.

Reply:
done